agent/xds/golden_test.go

```diff
@@ -124,12 +125,8 @@ func golden(t *testing.T, name, subname, latestSubname, got string) string {
 			return got
 		}
 
-		// We use require.JSONEq to compare values and ObjectsAreEqualValues is used
-		// internally by that function to compare the string JSON values. This only
-		// writes updates the golden file when they actually change.
-		if !assert.ObjectsAreEqualValues(gotInterface, latestExpectedInterface) {
```

Comment:
This line was actually added by @erichaberkorn to insulate us against the nondeterministic whitespace.